src/tribler/core/upgrade/tribler_db/decorator.py

```diff
@@ -0,0 +1,51 @@
+import functools
+import logging
+from typing import Callable
+
+from tribler.core.components.database.db.tribler_database import TriblerDatabase
+from tribler.core.utilities.pony_utils import db_session
+
+MIGRATION_METADATA = "_tribler_db_migration"
+
+logger = logging.getLogger('Migration (TriblerDB)')
+
+
+def migration(execute_only_if_version: int):
```

Comment:
**nitpick:** It looks correct, but it may be more convenient to specify the resulting version after the migration and assume the schema version before the migration should be one less. In that case, `@migration(schema_version=10)` would mean that the version after the migration is executed successfully should be 10, and the version before the migration should be 9. In that case, it will be a bit more obvious that after applying `@migration(schema_version=10)`, the resulting schema has version 10. But it is a matter of personal preferences.

Reply:
I agree that this is "a matter of personal preferences". Let's ask @xoriole for a second opinion.